peekingduck/pipeline/nodes/draw/utils/legend.py

```diff
@@ -27,52 +27,65 @@
     FILLED,
     PRIMARY_PALETTE,
     PRIMARY_PALETTE_LENGTH,
-    SMALL_FONTSCALE,
-    THICK,
     WHITE,
 )
 from peekingduck.pipeline.nodes.draw.utils.general import get_image_size
 
+ZONE_COUNTS_HEADING = "-ZONE COUNTS-"
 
-class Legend:
+
+class Legend:  # pylint: disable=too-many-instance-attributes
     """Legend class that uses available info to draw legend box on frame"""
 
-    def __init__(self) -> None:
-        self.legend_left_x = 15
+    def __init__(
+        self,
+        items: List[str],
+        position: str,
+        box_opacity: float,
+        font: Dict[str, Union[float, int]],
+    ) -> None:
+        self.items = items  # list of items to be drawn in legend box
+        self.position = position
+        self.box_opacity = box_opacity
+        self.font_size = font["size"]
+        self.font_thickness = font["thickness"]
 
         self.frame = None
+        self.legend_left_x = 15
         self.legend_starting_y = 0
         self.delta_y = 0
         self.legend_height = 0
+        self.item_height = cv2.getTextSize(
+            "",
+            FONT_HERSHEY_SIMPLEX,
+            self.font_size,
+            self.font_thickness,
+        )[0][1]
```

Comment:
It seems like this `getTextSize()` function has been called multiple time with only the `text` argument being different. And the return element used is always `[0][0/1]`. Consider writing a helper method for this

Reply:
Created helper function, also did the same for `cv2.putText()` which appears 3 times.